ui/src/pages/_document.js

```diff
@@ -16,25 +16,25 @@
 import Document, { Html, Head, Main, NextScript } from 'next/document';
 import { extractCritical } from '@emotion/server';
 
-export default class MyDocument extends Document {
-    static async getInitialProps(ctx) {
-        const initialProps = await Document.getInitialProps(ctx);
-        const styles = extractCritical(initialProps.html);
-        return {
-            ...initialProps,
-            styles: (
-                <>
-                    {initialProps.styles}
-                    <style
-                        data-emotion-css={styles.ids.join(' ')}
-                        nonce={ctx.req.headers.rid}
-                        dangerouslySetInnerHTML={{ __html: styles.css }}
-                    />
-                </>
-            ),
-        };
-    }
+export async function getServerSideProps(ctx) {
+    const initialProps = await Document.getInitialProps(ctx);
```

Comment:
Should this be Document.getServerSideProps?

Reply:
Document object provided by next doesn't have getServerSideProps.